### PR TITLE
Handle relative module replacements

### LINKED ISSUE #11: Time for a replaced dependency is NIL, causing a panic error
Hi! I have this `go.mod` file:
```go
require (
	github.com/Jeffail/gabs/v2 v2.5.1
	github.com/cenkalti/backoff/v4 v4.0.2
	github.com/cucumber/godog v0.9.0
	github.com/elastic/e2e-testing/cli v0.0.0-20200717181709-15d2db53ded7
	github.com/elastic/go-elasticsearch/v8 v8.0.0-20190731061900-ea052088db25
	github.com/sirupsen/logrus v1.4.2
)

replace github.com/elastic/e2e-testing/cli v0.0.0-20200717181709-15d2db53ded7 => ../cli
```
but when I run the tool with a Make target I have in the repository:
```make
 make notice
Generating NOTICE
go mod tidy
go mod download
go list -m -json all | go run go.elastic.co/go-licence-detector \
                -includeIndirect \
                -rules ../notice/rules.json \
                -overrides ../notice/overrides.json \
                -noticeTemplate ../notice/NOTICE.txt.tmpl \
                -noticeOut NOTICE.txt \
                -depsOut ""
```

I got this stacktrace:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1373dd1]

goroutine 1 [running]:
go.elastic.co/go-licence-detector/detector.mkDepInfo(0xc002a81950, 0xc0038121e0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/mdelapenya/sourcecode/pkg/mod/go.elastic.co/go-licence-detector@v0.4.0/detector/detector.go:213 +0x1c1
go.elastic.co/go-licence-detector/detector.doDetectLicences(0xc00cd71360, 0xc00000e4c0, 0xc00d190250, 0xc003be0200, 0x6, 0x8, 0xc0038121e0, 0x1fdefa0, 0xc000070060, 0xc0038125a0, ...)
        /Users/mdelapenya/sourcecode/pkg/mod/go.elastic.co/go-licence-detector@v0.4.0/detector/detector.go:152 +0xaf
go.elastic.co/go-licence-detector/detector.detectLicences(0xc00000e4c0, 0xc00d190250, 0xc0038125a0, 0xc0038121e0, 0x0, 0x0, 0x6abe7d8)
        /Users/mdelapenya/sourcecode/pkg/mod/go.elastic.co/go-licence-detector@v0.4.0/detector/detector.go:134 +0x97
go.elastic.co/go-licence-detector/detector.Detect(0x6abe7d8, 0xc0000717a0, 0xc00000e4c0, 0xc00d190250, 0xc0038121e0, 0x1, 0x18, 0xc000cce2fd, 0xc000171ed8)
        /Users/mdelapenya/sourcecode/pkg/mod/go.elastic.co/go-licence-detector@v0.4.0/detector/detector.go:102 +0xa0
main.main()
        /Users/mdelapenya/sourcecode/pkg/mod/go.elastic.co/go-licence-detector@v0.4.0/main.go:75 +0x327
exit status 2
make: *** [notice] Error 1
```

A sample PR and workspace can be found here: https://github.com/elastic/e2e-testing/pull/180

### PR DESCRIPTION
**Pull request by charith-elastic**
Relative module replacements don't have timestamps and cause nil pointer panics when trying to format time.

Fixes #11

### CODE REVIEW

**Comment by charith-elastic** [comment]
@mdelapenya can you run this patch on your project and confirm that it works? I'll tag a new release afterwards.

**Comment by mdelapenya** [comment]
Hey @charith-elastic, it runs properly, but the NOTICE.txt file it's not accurate. I'm pasting here the diff before and after the patch:

```diff
Dependency : github.com/elastic/e2e-testing/cli
- Version: v0.0.0-20200717181709-15d2db53ded7
+ Version: 
Licence type (autodetected): Elastic

- Contents of probable licence file $GOMODCACHE/github.com/elastic/e2e-testing/cli@v0.0.0-20200717181709-15d2db53ded7/LICENSE.txt:
+ Contents of probable licence file /Users/mdelapenya/sourcecode/src/github.com/elastic/e2e-testing/cli/LICENSE.txt:
```

File: detector/detector.go [hunk 1]
```diff
@@ -198,18 +198,27 @@ func mkDepInfo(mod *module, overrides dependency.Overrides) dependency.Info {
 	m := mod
 	if mod.Replace != nil {
 		m = mod.Replace
+		// If the replacement is a relative path, use the full path instead.
+		if strings.Contains(m.Path, "..") {
+			m.Path = mod.Path
```
Comment (mdelapenya) [inline detector/detector.go:203]:
```suggestion
			m.Path = mod.Path
			m.Version = mod.Version
```

**Comment by charith-elastic** [comment]
Ah, of course! This is another one of the reasons why I didn't add support for local replace directives in the first place. It's not quite as simple as it seems. I think I need to think about this a bit more and come up with a better solution.

**Comment by charith-elastic** [comment]
This won't work. I have listed the reasons in the linked issue.

**Pull request closed without merging**